### ISSUE TITLE
Builders, Deliverymen and Guards spin when they should be walking around or working.

### ISSUE THREAD

**Issue by lunadragnstar** [ldtteam/minecolonies#3161]
### Minecolonies version
Version: 4.54.1

### Expected behavior
The builders, deliverymen and guards all walking around like they should

### Actual behavior
Occasionally the builders, deliverymen and guards just stand and spin in circles. I've done some testing out o the builders. I notice they do this more when a request is put in at night when they are asleep, but not always. I've recalled them to their hut, to the townhall, fired them, rehired them, fired them and hired someone else, killed them and let another spawn. They all end up spinning. Sometimes they will stop if they sleep, but they sometimes start back up. What I've taken do doing is only ordering new things built during the day and if they work on something and start spinning, I cancel that order and restart it. This usually stops the spinning at least for a while. I'm playing on a single player, in creative, with the builders creating their own resources (changed the config file)

### Steps to reproduce the problem
I mostly see this in builders, I ask them to build something and they will start, then go to another spot and just spin. 

1. Order huts to be built
2. 
3. 
4. ... Profit

**Comment by kmarple1** [comment]
Same bug, version 0.10.77-ALPHA. In addition to spinning, my builder and deliverymen will sometimes just walk in a cycle, or get stuck against a wall. Picture of my builder trying to walk through the builder's hut attached. For spinning, knocking them away and placing a 2-high stack of blocks there usually works, but I've tried everything to break my builder's fascination with her back wall, with no luck.

![broken_builder](https://user-images.githubusercontent.com/8404056/50015948-77316e80-ff8d-11e8-8509-ae90fca4af8b.png)

**Comment by lunadragnstar** [comment]
I think I figured this out, they spin when they can't get to where they want to go, or at least can't figure out how to get to where they want to go.  I had one builder just spinning but when I broke a fence post he started working by going into where I broke the fence. Sometimes I just can't figure out where they want to go to fix it. This seems to happen a lot with the Dark Oak build or large builds like the Medieval Oak Town Hall.
